csep/utils/plots.py

```diff
@@ -568,14 +569,21 @@ def plot_basemap(basemap, extent, ax=None,  coastline=True, borders=False, linec
          linecolor (str): Color of borders and coast lines. default 'black',
          grid (bool): Draws a grid in the basemap
          grid_labels (bool): Annotate grid values
+         apprx (bool): If true, approximates transformation by setting aspect ratio of axes based on middle latitude
+         central_latitude (float): average latitude from plotting region
 
      Returns:
          :class:`matplotlib.pyplot.ax` object
 
      """
     if ax is None:
         fig = pyplot.figure()
-        ax = fig.add_subplot(111, projection=ccrs.PlateCarree(central_longitude=0.0))
+        ax = fig.add_subplot(111, projection=projection)
+        if apprx:
+            # Re-aspect plot (only for plain matplotlib plots, or when using PlateCarree)
+            LATKM = 110.574
+            ax.set_aspect(1 / (LATKM / 111.320 * numpy.cos(numpy.deg2rad(central_latitude))))
```

Comment:
I just noticed that it can be simplified:

    ax.set_aspect(111.320 * numpy.cos(numpy.deg2rad(central_latitude)) / LATKM)

Btw: here a reference for why/how this works: [stackoverflow.com/a/1253545](https://stackoverflow.com/a/1253545). I use it essentially for every geographic operation because it allows to go from lat-lon to relative distances in km in a very concise way.
So all in all, you may revise this section to:

            # Set plot aspect according to local longitude-to-latitude length ratio
            # (only compatible with plain PlateCarree "projection")
            LATKM = 110.574  # length of a ° of latitude [km] [https://stackoverflow.com/a/1253545]
            ax.set_aspect(111.320 * numpy.cos(numpy.deg2rad(central_latitude)) / LATKM)

Reply:
Damn. The simplification is wrong; it should be:

    ax.set_aspect(111.320 / LATKM / numpy.cos(numpy.deg2rad(central_latitude)))